### DISCUSSION TITLE
Is it to be expected that it takes multiple "make" statements to get a successful compile on a practically virgin install?

### DISCUSSION THREAD

**Discussion by fraternl** [Freetz-NG/freetz-ng#869] [Allgemein] [answered]
I can remember that when I previously did a "git clone https://github.com/Freetz-NG/freetz-ng.git" and then a "make menuconfig" and "make" it took some 5 to 10 "makes" after it got to a successful compile.

I'm still in the process of that and was wondering if it is normal.

If the answer is "yes", it's fine by me.
I merely want to know if it is to be expected and that there's nothing strange going on.

```
git clone https://github.com/Freetz-NG/freetz-ng.git
cd freetz-ng
make menuconfig
make
make
make
make
```

**Comment by fraternl** [comment]
It seems it doesn't end up in a successful make after all.
It's running on a Debian 11 inside a Hyper-V virtual machine.
This machine successfully compiled freetz-ng several times before and is not running out of diskspace.
It may be a memory problem. To run my old freetz environment simultaneously I had to change some of the settings which may have caused this.

I shut that one down now, but it seems I also need to turn the machine off/on to give it more memory (if that's the case).

```
make[3]: Nothing to be done for 'all'.
build/genautomata /home/freetz/freetz-ng/source/toolchain-mips_gcc-13.2.0_uClibc-1.0.44-nptl_kernel-4.9/gcc-13.2.0/gcc/common.md /home/freetz/freetz-ng/source/toolchain-mips_gcc-13.2.0_uClibc-1.0.44-nptl_kernel-4.9/gcc-13.2.0/gcc/config/mips/mips.md \
  insn-conditions.md > tmp-automata.cc
/bin/bash: line 2: 619881 Killed                  build/genautomata /home/freetz/freetz-ng/source/toolchain-mips_gcc-13.2.0_uClibc-1.0.44-nptl_kernel-4.9/gcc-13.2.0/gcc/common.md /home/freetz/freetz-ng/source/toolchain-mips_gcc-13.2.0_uClibc-1.0.44-nptl_kernel-4.9/gcc-13.2.0/gcc/config/mips/mips.md insn-conditions.md > tmp-automata.cc
make[3]: *** [Makefile:2505: s-automata] Error 137
make[2]: *** [Makefile:4603: all-gcc] Error 2
make[1]: *** [make/toolchain/target/gcc/gcc.mk:158: /home/freetz/freetz-ng/source/toolchain-mips_gcc-13.2.0_uClibc-1.0.44-nptl_kernel-4.9/gcc-13.2.0-initial/.compiled] Error 2
make: *** [Makefile:46: envira] Terminated
```

**Comment by fraternl** [comment]
I just took a physical machine I already had running with Debian 11 and was able to compile it in one go.
The machine is however not in my LAN, so it may need a complicated solution to transfer the image to the Fritzbox.

I now know the Hyper-V development machine is the culprit and will see if it has something to do with memory.

**Comment by fraternl** [comment, marked as answer]
I gave the virtual machine 2048 instead of 1024 RAM and turned it off/on
That didn't work.

I then shutdown the host machine, started the VM and it then compiled without problems.

"Have you tried turning it off and on again?"

**Comment by fda77** [comment]
I've a VPS with only 1GB ram and with this i'm also not able to build gcc13. A big swapfile/partitions helps, but makes it not very fast...
Set only 1 parallel procs did also not help

**Reply by fda77** [reply]
And yes, freetz/ has no gcc13 and no newer fritzos, and so it works with 1gb ram
